openmdao/core/implicitcomponent.py

```diff
@@ -26,6 +26,15 @@ def __init__(self, **kwargs):
         """
         super(ImplicitComponent, self).__init__(**kwargs)
 
+        if overrides_method('guess_nonlinear', self, ImplicitComponent):
```

Comment:
I need to recurse from the bottom up so that groups get their flag set, and this method goes from bottom up.